chain/chunks/src/lib.rs

```diff
@@ -1075,6 +1092,139 @@ impl ShardsManager {
         )
     }
 
+    /// Constructs a vector of [`ReceiptsProof`]s for specified chunk.
+    fn make_outgoing_receipts_proofs<T>(
+        &self,
+        chunk_header: &ShardChunkHeader,
+        outgoing_receipts: &Vec<Receipt>,
+        make_elem: impl Fn(ReceiptProof) -> T,
+    ) -> Result<Vec<T>, near_chunks_primitives::Error> {
+        let shard_id = chunk_header.shard_id();
+        let shard_layout = self
+            .runtime_adapter
+            .get_shard_layout_from_prev_block(chunk_header.prev_block_hash_ref())?;
+
+        let hashes = Chain::build_receipts_hashes(&outgoing_receipts, &shard_layout);
+        let (root, proofs) = merklize(&hashes);
+        if chunk_header.outgoing_receipts_root() != root {
+            return Err(near_chunks_primitives::Error::Other(format!(
+                "unexpected outgoing receipts root; want: {}, but got: {}",
+                chunk_header.outgoing_receipts_root(),
+                root
+            )));
```

Comment:
Yeah, I'd expect something like this here:

```suggestion
            return Err(near_chunks_primitives::Error::UnexpectedOutgoingRemeiptsRoot {
                want: chunk_header.outgoing_receipts_root(),
                got: root
            });
```